### PR TITLE
FIX: prevent usage of uninitialized message queue handle

### PR DESCRIPTION
**Pull request by RudiSV**
Signed-off-by: Rudolf Svanda <rudolf.svanda@globallogic.com>

### CODE REVIEW

**Comment by sebastianlipponer** [comment]
Note, that for `pthread_cond_wait()` spurious [wakeups may occur](https://linux.die.net/man/3/pthread_cond_wait). The current code does not account for this. Therefore, `dlt_segmented_queue_read_handle` might be uninitialized after `pthread_cond_wait` returns.

I think your change fixes the problem. However, I find your code hard to understand since it relies on the outer `while(1)` loop. What about replacing both `if()` conditions with a separate `while()` loop instead?

**Comment by RudiSV** [comment]
Fix does not target "spurious wakeups" even if it fix it too.
Problem occurs always if application exists and dlt_user.dlt_segmented_queue_read_handle is not initialized (message queue is not created)

Commit is updated.

**Comment by ssugiura** [comment]
@RudiSV I'm very sorry for replying late. The change itself looks good. Before merging to master, could you summarize the issue you faced and how you solved it in commit message so that others know what this commit improves? Thank you!

**Comment by RudiSV** [comment]
Hope it's better now.

**Comment by ssugiura** [comment]
@RudiSV Yes, it is! Thank you.
